### PR TITLE
Update changelog for 1.8.0

### PR DESCRIPTION
**Pull request by Egorand**

### CODE REVIEW

File: docs/changelog.md [hunk 1]
```diff
@@ -1,6 +1,25 @@
 Change Log
 ==========
 
+## Version 1.8.0
+
+_2021-03-22_
+
+ * New: Restructure interop artifacts, starting with `kotlinx-metadata`.
```
Comment (Egorand) [inline docs/changelog.md:8]: @ZacSweers want to expand on this a little more? Please feel free to push into the same branch.

Comment (ZacSweers) [inline docs/changelog.md:8]:
```suggestion
```

File: docs/changelog.md [hunk 2]
```diff
@@ -1,6 +1,25 @@
...
+ * New: Allow using a custom `ClassLoader` with `ReflectiveClassInspector`.
+ * New: Emit `abstract` modifier for abstract interface functions.
+ * Fix: Ensure `ImmutableKmProperty.toMutable()` copies `fieldSignature`.
+ * Fix: Prevent name clashes between an imported `MemberName` and a member in current scope.
+ * Fix: Prevent name clashes between a type and a supertype with the same name.
+ * Fix: Don't generate empty body for `expect` and `external` functions.
+ * Fix: Don't allow `expect` or `external` classes to initialize supertypes.
+ * Fix: Disallow delegate constructor calls in `external` classes.
+ * Fix: Allow non-public primary constructors inside inline/value classes.
+ * Fix: Allow init blocks inside inline/value classes.
```
Comment (ZacSweers) [inline docs/changelog.md:21]:
```suggestion
 * Fix: Allow init blocks inside inline/value classes.
 * Fix: Omit redundant `abstract` modifiers on members inside interfaces
```

**Review by ZacSweers — APPROVED** [review]

File: docs/changelog.md [hunk 3]
```diff
@@ -1,6 +1,25 @@
...
 
+## Version 1.8.0
+
+_2021-03-22_
+
+ 
+ * New: Kotlin 1.4.31.
+ * New: Add `KModifier.VALUE` to support `value class` declarations.
+ * New: Allow using a custom `ClassLoader` with `ReflectiveClassInspector`.
+ * New: Emit `abstract` modifier for abstract interface functions.
```
Comment (ZacSweers) [inline docs/changelog.md:12]:
```suggestion
 * New: Update to kotlinx-metadata 0.2.0.
```

**Pull request merged**